packages/mask/src/plugins/FindTruman/SNSAdaptor/PartsPanel.tsx

```diff
@@ -0,0 +1,328 @@
+                        disablePortal: true,
+                    }}
+                    arrow
+                    title={part.name}>
+                    <Typography noWrap variant="body2" color="text.primary">
+                        {part.name}
+                    </Typography>
+                </Tooltip>
+            </Box>
+        </div>
+    )
+}
+
+function QuestItem(props: { quest: Quest }) {
+    const { quest } = props
+    const { classes, cx } = useStyles()
+    const { t, const: consts } = useContext(FindTrumanContext)
+
+    const { open: isQuestDialogOpen, onOpen: onQuestDialogOpen, onClose: onQuestDialogClose } = useControlledDialog()
+
+    return (
+        <div>
+            <Typography textAlign="center" variant="body2" color="text.secondary" gutterBottom>
+                {formatDateTime(new Date(quest.startFrom), 'yyyy-MM-dd')}
+            </Typography>
+            <div className={classes.ribbonWrapper}>
+                <img className={classes.cover} src={consts?.boxImg} />
+                <div className={cx(classes.ribbon, classes.ribbonGray)}>
+                    <Typography className={cx(classes.ribbonContent, classes.ribbonContentGray)} variant="body2">
+                        {t('plugin_find_truman_dialog_ribbon_lacked')}
+                    </Typography>
+                </div>
+            </div>
+            <Button size="small" color="primary" variant="outlined" fullWidth onClick={onQuestDialogOpen}>
+                {t('plugin_find_truman_dialog_get')}
+            </Button>
+            <QuestDialog quest={quest} open={isQuestDialogOpen} onClose={onQuestDialogClose} />
+        </div>
+    )
+}
+
+function MysteryBoxItem(props: { account: string; box: MysteryBox; onOpened: () => void }) {
+    const { account, box, onOpened } = props
+    const { classes } = useStyles()
+    const { t, const: consts } = useContext(FindTrumanContext)
+    const [loading, setLoading] = useState(box.isOpened && !box.isMinted)
+
+    const startPolling = () => {
+        const polling = setInterval(async () => {
+            try {
+                const boxInfo = await fetchMysteryBoxInfo(box.id)
+                if (boxInfo.isMinted) {
+                    polling && clearInterval(polling)
+                    setLoading(false)
+                    onOpened()
+                }
+            } catch (error) {
+                polling && clearInterval(polling)
+                setLoading(false)
+                onOpened()
+            }
+        }, 2000)
+        return polling
+    }
+
+    const handleOpenBox = async () => {
+        try {
+            setLoading(true)
+            await openMysteryBox(account, {
+                timestamp: getUnixTime(Date.now()),
+                address: account,
+                rawdata: {
+                    boxId: box.id,
+                },
+            })
+            startPolling()
+        } catch (error) {
+            onOpened()
+            setLoading(false)
```

Comment:
It will not reset the `loading` if everything goes fine. Do you need to move this in the `finally` block?